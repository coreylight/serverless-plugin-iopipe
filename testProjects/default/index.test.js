import _ from 'lodash';
import AdmZip from 'adm-zip';

test('Generated file requires plugin and includes plugin inline', async () => {
  const zip = new AdmZip('./.serverless/sls-unit-test-default.zip');
  expect(1).toBe(1);
  const handlerFile = _.find(
    zip.getEntries(),
    entry => entry.entryName === 'iopipe-handlers.js'
  );
  const fileContents = handlerFile.getData().toString('utf8');
  expect(fileContents).toMatchSnapshot();
  /*eslint-disable no-eval*/
  eval(fileContents);
  const result = await new Promise(succeed => {
    exports.simple({}, { succeed });
  });
  expect(result.statusCode).toEqual(200);
});
