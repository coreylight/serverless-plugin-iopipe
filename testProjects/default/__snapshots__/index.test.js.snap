// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generated file requires plugin and includes plugin inline 1`] = `
"const iopipe = require('iopipe')({\\"token\\":\\"test-token\\"});

exports['simple'] = function attemptSimple0(event, context, callback) {
  try {
    return iopipe((evt, ctx, cb) => {
      return require('./handlers/simple').handler(evt, ctx, cb);
    })(event, context, callback);
  } catch (err) {
    throw err;
  }
};

exports['multiple'] = function attemptMultiple1(event, context, callback) {
  try {
    return iopipe((evt, ctx, cb) => {
      return require('./handlers/multiple').handler(evt, ctx, cb);
    })(event, context, callback);
  } catch (err) {
    throw err;
  }
};

exports['multipleDifferentHandler'] = function attemptMultipleDifferentHandler2(event, context, callback) {
  try {
    return iopipe((evt, ctx, cb) => {
      return require('./handlers/multiple').differentNameHandler(evt, ctx, cb);
    })(event, context, callback);
  } catch (err) {
    throw err;
  }
};

exports['es5'] = function attemptEs53(event, context, callback) {
  try {
    return iopipe((evt, ctx, cb) => {
      return require('./handlers/es5').handler(evt, ctx, cb);
    })(event, context, callback);
  } catch (err) {
    throw err;
  }
};

exports['multiple-dots-in-name'] = function attemptMultipleDotsInName4(event, context, callback) {
  try {
    return iopipe((evt, ctx, cb) => {
      return require('./handlers/multiple.dots.in.name').handler(evt, ctx, cb);
    })(event, context, callback);
  } catch (err) {
    throw err;
  }
};

exports['noModule'] = function attemptNoModule5(event, context, callback) {
  try {
    return iopipe((evt, ctx, cb) => {
      return require('./handlers/noModule').handler(evt, ctx, cb);
    })(event, context, callback);
  } catch (err) {
    throw err;
  }
};

exports['syntaxError'] = function attemptSyntaxError6(event, context, callback) {
  try {
    return iopipe((evt, ctx, cb) => {
      return require('./handlers/syntaxError').handler(evt, ctx, cb);
    })(event, context, callback);
  } catch (err) {
    throw err;
  }
};

exports['es5Named'] = function attemptEs5Named7(event, context, callback) {
  try {
    return iopipe((evt, ctx, cb) => {
      return require('./handlers/es5Named').handler(evt, ctx, cb);
    })(event, context, callback);
  } catch (err) {
    throw err;
  }
};
"
`;
